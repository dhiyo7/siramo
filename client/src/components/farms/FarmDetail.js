import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Image
} from 'react-native'
import { inject, observer } from 'mobx-react'
import UserStore from '../../store/UserStore'
import FarmStore from '../../store/FarmStore'

import CircleProg from '../customs/CircleProg'
import TemperatureGauge from '../customs/TemperatureGauge'
import { Ionicons } from '@expo/vector-icons'
import FarmPic from '../farms/FarmsPic'
import FarmDate from '../farms/FarmDate'
import ButtonOnOff from '../farms/ButtonOnOff'

@observer class FarmDetail extends Component {
  render() {
    const {
      name, temperature, water_ratio, ready_siram,
      water_level, humidity, last_siram, last_updated,
    } = FarmStore.FarmDetail
    const { dateFormat, timeFormat } = UserStore

    return (
      <View style={styles.mainContainer}>
        <FarmPic />
        <View style={styles.DetailCard}>
          <View style={styles.infoView}>
            <Text style={styles.textTitle}>{name}</Text>
          </View>
          <TemperatureGauge
            sensor={temperature}
            textTitle={'Temperature'}
            maxValue={'Â°C'}
            color='rgba(241, 6, 102, 0.83)'
          />
          <CircleProg 
            sensor={water_ratio}
            textTitle={'Water Ratio'}
            maxValue={'%'}
            color='rgba(0, 167, 5, 1)' 
          />
          <CircleProg 
            sensor={(water_level/50)*100}
            textTitle={'Water Level'}
            maxValue={'%'}
            color='rgb(41, 177, 237)' 
          />
          <CircleProg 
            sensor={humidity}
            textTitle={'Humidity'}
            maxValue={'%'}
            color='rgba(242, 129, 35, 0.83)'
          />
          <View style={styles.dateContainer}>
              <FarmDate 
                date={{
                  title: 'Last Watering:',
                  dates: dateFormat(last_siram),
                  times: timeFormat(last_siram)
                }}
              />
              <FarmDate 
                date={{
                  title: 'Last Updated:',
                  dates: dateFormat(last_updated),
                  times: timeFormat(last_updated)
                }}
              />
          </View>
          {/* <Text>Ready Siram: {ready_siram}</Text> */}
        </View>
        <ButtonOnOff siram={{ready_siram, triggerSiram: FarmStore.triggerSiram}}/>
      </View>
    )
  }
} 

const styles = StyleSheet.create({
  mainContainer: {
    alignItems: 'center',
    width: '100%',
    backgroundColor: 'white'
  },
  DetailCard: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    backgroundColor: '#fff',
    padding: 4
  },
  textTitle: {
    fontSize: 20,
    textAlign: 'center',
    width: '100%'
  },
  dateView: {
    width: '100%',
    height: 'auto',
    borderRadius: 10
  },
  dateText: {
    textAlign: 'center',
    fontSize: 20
  },
  infoView: {
    width: '100%',
  },
  infoText: {
    fontSize: 20 
  },
  dateContainer: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    width: '100%',
    height: 'auto',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center'
  }
})

export default FarmDetail