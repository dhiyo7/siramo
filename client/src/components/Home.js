import React, { Component } from 'react'
import {
  View, Text, StyleSheet
} from 'react-native'

// Sementara
import FarmDetail from './farms/FarmDetail'

class Home extends Component {
  render() {
    return (
      <View style={styles.mainContainer}>
        <Text> Halaman Home ini ? </Text>
        <FarmDetail />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: '#151e2d',
    alignItems: 'center',
    justifyContent: 'flex-start',
  }
})

export default Home