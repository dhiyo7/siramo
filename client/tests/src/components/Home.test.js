import React from 'react'
import { View, ScrollView } from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Home from '../../../src/components/Home'
import FarmDetail from '../../../src/components/farms/FarmDetail'

Enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<Home />)
})

describe('<Home /> snapshot Testing', () => {
  it('snapshot works!', () => {
    const tree = renderer.create(<Home />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<Home />', () => {
  it('should render <Home />', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<Home /> child is rendered', () => {
  it('should render <Home /> child Element', () => {
    expect(wrapper.find(<ScrollView />)).toBeDefined()
    expect(wrapper.find(<View />)).toBeDefined()
    expect(wrapper.find(<FarmDetail />)).toBeDefined()
    expect(wrapper.find('FarmDetail').length).toBe(1)
  })
})

// componentWillMount sama componentwillUnmount belum di test