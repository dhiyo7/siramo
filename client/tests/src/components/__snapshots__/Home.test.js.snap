// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> snapshot Testing snapshot works! 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1F8E9",
          "flex": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "margin": 10,
              "paddingTop": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "http://jurnalaquascape.com/wp-content/uploads/2015/05/amazon-sword.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 75,
                "height": 150,
                "width": 150,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F1F8E9",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "padding": 4,
            }
          }
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "margin": 12,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <ARTSurfaceView
                style={
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    },
                    Object {
                      "height": 120,
                      "width": 120,
                    },
                  ]
                }
              >
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      6,
                      2,
                      60,
                      6,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.9450980392156862,
                      0.023529411764705882,
                      0.4,
                      0.83,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={10}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      0.5,
                      4,
                      60,
                      60,
                      59.5,
                      -1.5707963267948966,
                      1.5707963267948966,
                      1,
                      4,
                      60,
                      60,
                      59.5,
                      1.5707963267948966,
                      -1.5707963267948966,
                      1,
                      1,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.9450980392156862,
                      0.023529411764705882,
                      0.4,
                      0.83,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={1}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
              </ARTSurfaceView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 49,
                    "height": 98,
                    "justifyContent": "center",
                    "left": 11,
                    "position": "absolute",
                    "top": 11,
                    "width": 98,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(241, 6, 102, 0.83)",
                        "fontSize": 21.77777777777778,
                        "fontWeight": "300",
                      },
                      Object {
                        "fontSize": 16,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Temperature
0Â°C
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "margin": 12,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <ARTSurfaceView
                style={
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    },
                    Object {
                      "height": 120,
                      "width": 120,
                    },
                  ]
                }
              >
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      6,
                      2,
                      60,
                      6,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0,
                      0.6549019607843137,
                      0.0196078431372549,
                      1,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={10}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      0.5,
                      4,
                      60,
                      60,
                      59.5,
                      -1.5707963267948966,
                      1.5707963267948966,
                      1,
                      4,
                      60,
                      60,
                      59.5,
                      1.5707963267948966,
                      -1.5707963267948966,
                      1,
                      1,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0,
                      0.6549019607843137,
                      0.0196078431372549,
                      1,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={1}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
              </ARTSurfaceView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 49,
                    "height": 98,
                    "justifyContent": "center",
                    "left": 11,
                    "position": "absolute",
                    "top": 11,
                    "width": 98,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 167, 5, 1)",
                        "fontSize": 21.77777777777778,
                        "fontWeight": "300",
                      },
                      Object {
                        "fontSize": 16,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Water Ratio
0%
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "margin": 12,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <ARTSurfaceView
                style={
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    },
                    Object {
                      "height": 120,
                      "width": 120,
                    },
                  ]
                }
              >
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      6,
                      2,
                      60,
                      6,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.1607843137254902,
                      0.6941176470588235,
                      0.9294117647058824,
                      1,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={10}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      0.5,
                      4,
                      60,
                      60,
                      59.5,
                      -1.5707963267948966,
                      1.5707963267948966,
                      1,
                      4,
                      60,
                      60,
                      59.5,
                      1.5707963267948966,
                      -1.5707963267948966,
                      1,
                      1,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.1607843137254902,
                      0.6941176470588235,
                      0.9294117647058824,
                      1,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={1}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
              </ARTSurfaceView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 49,
                    "height": 98,
                    "justifyContent": "center",
                    "left": 11,
                    "position": "absolute",
                    "top": 11,
                    "width": 98,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgb(41, 177, 237)",
                        "fontSize": 21.77777777777778,
                        "fontWeight": "300",
                      },
                      Object {
                        "fontSize": 16,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Water Level
0%
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "margin": 12,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <ARTSurfaceView
                style={
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "0deg",
                        },
                      ],
                    },
                    Object {
                      "height": 120,
                      "width": 120,
                    },
                  ]
                }
              >
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      6,
                      2,
                      60,
                      6,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.9490196078431372,
                      0.5058823529411764,
                      0.13725490196078433,
                      0.83,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={10}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
                <ARTShape
                  d={
                    Array [
                      0,
                      60,
                      0.5,
                      4,
                      60,
                      60,
                      59.5,
                      -1.5707963267948966,
                      1.5707963267948966,
                      1,
                      4,
                      60,
                      60,
                      59.5,
                      1.5707963267948966,
                      -1.5707963267948966,
                      1,
                      1,
                    ]
                  }
                  fill={null}
                  opacity={1}
                  stroke={
                    Array [
                      0.9490196078431372,
                      0.5058823529411764,
                      0.13725490196078433,
                      0.83,
                    ]
                  }
                  strokeCap={0}
                  strokeDash={null}
                  strokeJoin={1}
                  strokeWidth={1}
                  transform={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                />
              </ARTSurfaceView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 49,
                    "height": 98,
                    "justifyContent": "center",
                    "left": 11,
                    "position": "absolute",
                    "top": 11,
                    "width": 98,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "rgba(242, 129, 35, 0.83)",
                        "fontSize": 21.77777777777778,
                        "fontWeight": "300",
                      },
                      Object {
                        "fontSize": 16,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Humidity
0%
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderRadius": 10,
                "height": "auto",
                "width": "100%",
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Last Watering:
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#128eb7",
                    "borderColor": "#ddd",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "elevation": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.8,
                    "shadowRadius": 2,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 35,
                      "margin": 8,
                      "padding": 4,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 25,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                     
                    01 January 1970
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 35,
                      "margin": 8,
                      "padding": 4,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 25,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                     
                    07:00
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderRadius": 10,
                "height": "auto",
                "width": "100%",
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Last updated:
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#128eb7",
                    "borderColor": "#ddd",
                    "borderRadius": 2,
                    "borderWidth": 1,
                    "elevation": 1,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "center",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.8,
                    "shadowRadius": 2,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 35,
                      "margin": 8,
                      "padding": 4,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 25,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                     
                    01 January 1970
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 35,
                      "margin": 8,
                      "padding": 4,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 25,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                     
                    07:00
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#31ce12",
              "borderRadius": 4,
              "marginBottom": 4,
              "marginTop": 8,
              "opacity": 1,
              "paddingVertical": 15,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
             
            Start Watering
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
